src/Propulsion/Ingestion.fs

```diff
@@ -127,7 +127,9 @@ type Ingester<'Items> private
             while applyIncoming handleIncoming || applyMessages stats.Handle do ()
             stats.RecordCycle()
             if stats.Interval.IfDueRestart() then let struct (active, max) = maxRead.State in stats.DumpStats(active, max)
-            do! Task.WhenAny(awaitIncoming ct, awaitMessage ct, Task.Delay(stats.Interval.RemainingMs)) :> Task }
+            let cts = CancellationTokenSource.CreateLinkedTokenSource(ct)
+            do! Task.WhenAny(awaitIncoming cts.Token, awaitMessage cts.Token, Task.Delay(stats.Interval.RemainingMs, cts.Token)) :> Task
```

Comment:
I thought about generalising this as `Task.race` but each of these has a different return type so it would involve some extra casts to `Task` so decided against it

Reply:
Maybe a helper `Task.runWithCancellation` that has an `InlineIfLambda` arg that gets passed the `ct.Token`. Would enable a 2 line comment explaining the critical nature of doing the cancel ?